src/test/java/io/cryostat/net/TargetConnectionManagerTest.java

```diff
@@ -210,4 +210,74 @@ public JFRConnection answer(InvocationOnMock invocation)
         JFRConnection conn2 = mgr.executeConnectedTask(desc, a -> a);
         MatcherAssert.assertThat(conn1, Matchers.not(Matchers.sameInstance(conn2)));
     }
+
+    @Test
+    void shouldCreateNewConnectionWhenMaxSizeZeroed() throws Exception {
+        TargetConnectionManager mgr =
+                new TargetConnectionManager(
+                        () -> jfrConnectionToolkit, Duration.ofSeconds(10), 0, logger);
+        Mockito.when(jfrConnectionToolkit.createServiceURL(Mockito.anyString(), Mockito.anyInt()))
+                .thenAnswer(
+                        new Answer<JMXServiceURL>() {
+                            @Override
+                            public JMXServiceURL answer(InvocationOnMock args) throws Throwable {
+                                String host = args.getArgument(0);
+                                int port = args.getArgument(1);
+                                return new JMXServiceURL(
+                                        "rmi",
+                                        "",
+                                        0,
+                                        String.format("/jndi/rmi://%s:%d/jmxrmi", host, port));
+                            }
+                        });
+        Mockito.when(jfrConnectionToolkit.connect(Mockito.any(), Mockito.any(), Mockito.any()))
+                .thenAnswer(
+                        new Answer<JFRConnection>() {
+                            @Override
+                            public JFRConnection answer(InvocationOnMock invocation)
+                                    throws Throwable {
+                                return Mockito.mock(JFRConnection.class);
+                            }
+                        });
+        ConnectionDescriptor desc = new ConnectionDescriptor("foo");
+        JFRConnection conn1 = mgr.executeConnectedTask(desc, a -> a);
+        Thread.sleep(10);
```

Comment:
I wasn't sure how a `Caffeine` cache behaves when the maximum size is zeroed out so I looked it up and the docs say the connection will be immediately evicted upon being loaded into the cache. In that case, I could be mistaken but I don't think it's necessary to sleep the thread for the TTL duration because a new connection should be created right away, no?

Reply:
It will be "immediately" evicted, but this depends on the Caffeine/system scheduler, which does eviction and other tasks like this asynchronously on a worker thread. If I don't add a sleep here then the scheduler doesn't actually have a chance to evict the connection from the cache before I grab it again. That's what the [docs](https://javadoc.io/doc/com.github.ben-manes.caffeine/caffeine/latest/com.github.benmanes.caffeine/com/github/benmanes/caffeine/cache/Caffeine.html) mean by:

> As eviction is scheduled on the configured executor, tests may instead prefer to configure the cache to execute tasks directly on the same thread. 

I could, and maybe should, make it so that the tests use a same-thread scheduler, rather than the system scheduler, and that would make this test deterministic and remove the need for this sleep.

Reply:
I'll approve the current changes in the meantime then.

Reply:
I just added a commit that configures the cache to use the current thread and no background scheduling for the unit tests where this makes sense. Otherwise the default executor (ForkJoinPool) and the system scheduler thread are still used.